documentation/src/main/asciidoc/public/reference/_indexing-plan.adoc

```diff
@@ -113,10 +113,17 @@ include::../components/_writing-reading-intro-note.adoc[]
 
 [WARNING]
 ====
-Indexing synchronization is only relevant when <<coordination-none,coordination is disabled>>.
-
-With coordination strategies such as <<coordination-outbox-polling,`outbox-polling`>>,
-indexing happens in background threads and is always asynchronous.
+No matter if the <<coordination-none,coordination is disabled>> or a coordination strategy such as <<coordination-outbox-polling,`outbox-polling`>>
+is used, when it comes to indexing, Hibernate Search relies on indexing plans.
+Depending on the coordination strategy a plan resulting in index changes may be constructed within the same
+session, where the entities have been modified (when <<coordination-none,coordination is disabled>>),
+or with an entirely different session in some background thread (with <<coordination-outbox-polling,`outbox-polling`>> enabled).
+
+In the first scenario it makes sense to control how the indexing plan synchronization happens,
+since it has a direct impact on the application thread behavior.
+While in the second one, indexing happens in background threads and is always asynchronous.
+Because of that, setting a non-default indexing plan synchronization strategy, with the
+<<coordination-outbox-polling,`outbox-polling`>> coordination being enabled, will lead to an exception at the startup.
```

Comment:
This is probably a bit too long for a warning: if we want people to read it, we need to keep it short and to the point :D 

```suggestion
When using the <<coordination-outbox-polling,`outbox-polling` coordination strategy>>,
the actual indexing plan performing the index changes
is created asynchronously in a background thread.

Because of that, with that coordination strategy it does not make sense
to set a non-default indexing plan synchronization strategy,
and doing so will lead to an exception on startup.
```